tests/Unit/Domain/DomainCreators/Test_Disk.cpp

```diff
@@ -0,0 +1,167 @@
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Disk.Boundaries",
+                  "[Domain][Unit]") {
+  const double inner_radius = 1.0, outer_radius = 2.0;
+  const size_t refinement_level = 2;
+  const std::array<size_t, 2> grid_points{{4, 4}};
+
+  const DomainCreators::Disk disk{inner_radius, outer_radius, refinement_level,
+                                  grid_points};
+  test_physical_separation(disk.create_domain().blocks());
+  const Orientation<2> aligned_orientation{};
+  const Orientation<2> quarter_turn_ccw(std::array<Direction<2>, 2>{
+      {Direction<2>::lower_eta(), Direction<2>::upper_xi()}});
+  const Orientation<2> half_turn(std::array<Direction<2>, 2>{
+      {Direction<2>::lower_xi(), Direction<2>::lower_eta()}});
+  const Orientation<2> quarter_turn_cw(std::array<Direction<2>, 2>{
+      {Direction<2>::upper_eta(), Direction<2>::lower_xi()}});
+  test_disk_construction(
+      disk, inner_radius, outer_radius,
+      {grid_points,
+       grid_points,
+       grid_points,
+       grid_points,
+       {{grid_points[1], grid_points[1]}}},
+      {5, make_array<2>(refinement_level)},
+      std::vector<std::unordered_map<Direction<2>, BlockNeighbor<2>>>{
+          {{Direction<2>::lower_eta(), {3, aligned_orientation}},
+           {Direction<2>::upper_eta(), {1, aligned_orientation}},
+           {Direction<2>::lower_xi(), {4, aligned_orientation}}},
+          {{Direction<2>::lower_eta(), {0, aligned_orientation}},
+           {Direction<2>::upper_eta(), {2, aligned_orientation}},
+           {Direction<2>::lower_xi(), {4, quarter_turn_cw}}},
+          {{Direction<2>::lower_eta(), {1, aligned_orientation}},
+           {Direction<2>::upper_eta(), {3, aligned_orientation}},
+           {Direction<2>::lower_xi(), {4, half_turn}}},
+          {{Direction<2>::lower_eta(), {2, aligned_orientation}},
+           {Direction<2>::upper_eta(), {0, aligned_orientation}},
+           {Direction<2>::lower_xi(), {4, quarter_turn_ccw}}},
+          {{Direction<2>::upper_xi(), {0, aligned_orientation}},
+           {Direction<2>::upper_eta(), {1, quarter_turn_ccw}},
+           {Direction<2>::lower_xi(), {2, half_turn}},
+           {Direction<2>::lower_eta(), {3, quarter_turn_cw}}}},
+      std::vector<std::unordered_set<Direction<2>>>{
+          {{Direction<2>::upper_xi()}},
+          {{Direction<2>::upper_xi()}},
+          {{Direction<2>::upper_xi()}},
+          {{Direction<2>::upper_xi()}},
+          {}});
+}
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Disk.Factory", "[Domain][Unit]") {
+  const auto disk = test_factory_creation<DomainCreator<2, Frame::Inertial>>(
+      "  Disk:\n"
+      "    InnerRadius: 1\n"
+      "    OuterRadius: 3\n"
+      "    InitialRefinement: 4\n"
+      "    InitialGridPoints: [2,3]\n");
+
+  const double inner_radius = 1.0, outer_radius = 3.0;
+  const size_t refinement_level = 4;
+  const std::array<size_t, 2> grid_points{{2, 3}};
```

Comment:
move the common parts into test_disk_construction and just pass in the parameters taken from the factory string